src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -1017,6 +1166,18 @@ consume_low_fd(void)
 			DS_ERROR(errno, "failed to reserve a low fd");
 			goto err;
 		} else if (low_fd_list[low_fd_count] >= DAOS_MIN_FD) {
+			/* reserve fd 255 too. 255 is commonly used by bash. */
+			rc = dup2(low_fd_list[0], 255);
+			if (rc == -1) {
+				DS_ERROR(errno, "dup2() failed to reserve fd 255");
+				goto err;
+			}
+			/* reserve fd DAOS_DUMMY_FD which will be used later by dup2(). */
+			rc = dup2(low_fd_list[0], DAOS_DUMMY_FD);
```

Comment:
Same comment as above I think, this will close fd 1001.  That's unlikely to be in use but you can't hard-code fds and then just start using them, that's the exact class of bug that exists in bash scripts that this code is trying to work around.

Reply:
Replaced dup2() with fcntl() as you suggested. Thank you!